src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -289,9 +336,18 @@ public JCTree translateTopLevelClass(Env<AttrContext> env, JCTree cdef, TreeMake
     @Override
     public void visitClassDef(JCClassDecl tree) {
         KlassInfo prevKlassInfo = kInfo;
```

Comment:
future improvement? we have class info, lambda context, capture site info, should we probably gather them all in an environment like structure?

Reply:
yes and no. First, note that capture site info is gone.
As for lambda context and klass info, we use them in different ways: the lambda context is set when we're in a lambda. The klass info is set when we are inside a klass. We create many lambda contexts for each klass info. We could of course move the lambda context field from the visitor to klass info, but I'm not sure how much that buys.